packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/models/MethodParameterInfoModel.java

```diff
@@ -49,6 +49,13 @@ public Class<MethodParameterInfoModel> getCommonModelClass() {
 
     public abstract int getModifiers();
 
+    /**
+     * Returns a zero-based index of the parameter in the method parameter list.
+     * 
+     * @return the index, from 0 to n-1
+     */
+    public abstract int getIndex();
```

Comment:
I would suggest using the same memoization approach as used for other properties. It would also allow constructing the schema if necessary.